packages/eslint-plugin-circuit-ui/no-invalid-custom-properties/index.ts

```diff
@@ -0,0 +1,230 @@
+  'bg-warning-disabled',
+  'bg-warning-strong',
+  'bg-warning-strong-hovered',
+  'bg-warning-strong-pressed',
+  'bg-warning-strong-disabled',
+  /* Danger backgrounds */
+  'bg-danger',
+  'bg-danger-hovered',
+  'bg-danger-pressed',
+  'bg-danger-disabled',
+  'bg-danger-strong',
+  'bg-danger-strong-hovered',
+  'bg-danger-strong-pressed',
+  'bg-danger-strong-disabled',
+  /* Promo backgrounds */
+  'bg-promo',
+  'bg-promo-hovered',
+  'bg-promo-pressed',
+  'bg-promo-disabled',
+  'bg-promo-strong',
+  'bg-promo-strong-hovered',
+  'bg-promo-strong-pressed',
+  'bg-promo-strong-disabled',
+  /* Neutral foregrounds */
+  'fg-normal',
+  'fg-normal-hovered',
+  'fg-normal-pressed',
+  'fg-normal-disabled',
+  'fg-subtle',
+  'fg-subtle-hovered',
+  'fg-subtle-pressed',
+  'fg-subtle-disabled',
+  'fg-placeholder',
+  'fg-placeholder-hovered',
+  'fg-placeholder-pressed',
+  'fg-placeholder-disabled',
+  'fg-on-strong',
+  'fg-on-strong-hovered',
+  'fg-on-strong-pressed',
+  'fg-on-strong-disabled',
+  /* Accent foregrounds */
+  'fg-accent',
+  'fg-accent-hovered',
+  'fg-accent-pressed',
+  'fg-accent-disabled',
+  /* Success foregrounds */
+  'fg-success',
+  'fg-success-hovered',
+  'fg-success-pressed',
+  'fg-success-disabled',
+  /* Warning foregrounds */
+  'fg-warning',
+  'fg-warning-hovered',
+  'fg-warning-pressed',
+  'fg-warning-disabled',
+  /* Danger foregrounds */
+  'fg-danger',
+  'fg-danger-hovered',
+  'fg-danger-pressed',
+  'fg-danger-disabled',
+  /* Promo foregrounds */
+  'fg-promo',
+  'fg-promo-hovered',
+  'fg-promo-pressed',
+  'fg-promo-disabled',
+  /* Neutral borders */
+  'border-normal',
+  'border-normal-hovered',
+  'border-normal-pressed',
+  'border-normal-disabled',
+  'border-subtle',
+  'border-subtle-hovered',
+  'border-subtle-pressed',
+  'border-subtle-disabled',
+  'border-divider',
+  'border-divider-hovered',
+  'border-divider-pressed',
+  'border-divider-disabled',
+  'border-strong',
+  'border-strong-hovered',
+  'border-strong-pressed',
+  'border-strong-disabled',
+  /* Accent borders */
+  'border-accent',
+  'border-accent-hovered',
+  'border-accent-pressed',
+  'border-accent-disabled',
+  /* Success borders */
+  'border-success',
+  'border-success-hovered',
+  'border-success-pressed',
+  'border-success-disabled',
+  /* Warning borders */
+  'border-warning',
+  'border-warning-hovered',
+  'border-warning-pressed',
+  'border-warning-disabled',
+  /* Danger borders */
+  'border-danger',
+  'border-danger-hovered',
+  'border-danger-pressed',
+  'border-danger-disabled',
+  /* Promo borders */
+  'border-promo',
+  'border-promo-hovered',
+  'border-promo-pressed',
+  'border-promo-disabled',
+  /* Special colors */
+  'bg-overlay',
+  'bg-elevated',
+  'border-focus',
+];
+
+/* eslint-disable no-useless-escape, @typescript-eslint/quotes */
+const REGEX_STRING = `(?:--cui-)(?!(?:${VALID_CUSTOM_PROPERTIES_WITHOUT_PREFIX.join(
+  '|',
+)})[^\w\-])[a-zA-Z0-9\-]+`;
```

Comment:
Yeah sure, thanks CodeQL 🤖 I just think it's a bit more readable for a human because `-` is not _always_ matching an actual unicode dash. For example in `a-z` the dash represents something else entirely.

So we end up with:
```js
const atoz = /[a-z]/; // matches a letter character from a to z
const adash = /[a-]/; // matches the "a" and "-" unicode characters
```

I like the escaping because it makes it explicit that we want to match the "-" character (since custom properties are usually things like `--my-custom-property`)